From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Glenn Baker <glenn.baker@gmx.com>
Date: Tue, 7 Mar 2023 14:51:52 +0100
Subject: [PATCH] 6011: AVR8: Fix DES operation reconstruction.

The AVR Instruction Set Manual defines the DES operation as follows:

If H = 0 then Encrypt round (R7-R0, R15-R8, K)
If H = 1 then Decrypt round (R7-R0, R15-R8, K)

Signed-off-by: Glenn Baker <glenn.baker@gmx.com>
---
 Ghidra/Processors/Atmel/data/languages/avr8.sinc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Ghidra/Processors/Atmel/data/languages/avr8.sinc b/Ghidra/Processors/Atmel/data/languages/avr8.sinc
index c7e066154..90a39754f 100644
--- a/Ghidra/Processors/Atmel/data/languages/avr8.sinc
+++ b/Ghidra/Processors/Atmel/data/languages/avr8.sinc
@@ -739,7 +739,7 @@ define pcodeop decrypt;
 
 :des op4to7                      is phase=1 & ophi8=0x94 & oplow4=0xb & op4to7 {
 	val:1  = op4to7;
-	if (Hflg) goto <enc>;
+	if (Hflg == 0) goto <enc>;
 	R15R14R13R12R11R10R9R8 = decrypt(R7R6R5R4R3R2R1R0, val);
 	goto inst_next;
   <enc>
-- 
2.43.0

